src/odemis/driver/lakeshore.py

```diff
@@ -0,0 +1,510 @@
+                    output_type,
+                    heater_resistance,
+                    max_current,
+                    max_user_current,
+                    current_or_power))
+
+    def GetHeaterSetup(self):
+        """
+        Query heater setup from device
+
+        returns: tuple of ints with the setup parameters in sequence
+        """
+        htr_setup = self._sendQuery(b'HTRSET? %d' % (self._output_channel,))
+        (output_type,
+            heater_resistance,
+            max_current,
+            max_user_current,
+            current_or_power) = htr_setup.split(',')
+
+        return (int(output_type),
+            int(heater_resistance),
+            int(max_current),
+            int(max_user_current),
+            int(current_or_power))
+
+    def SetHeaterRange(self, hrange):
+        """
+        Set the heater range
+        hrange (int): For Outputs 1 and 2 in Current mode: 0 = Off, 1 = Low, 
+            2 = Medium, 3 = High
+            For Output 2 in Voltage mode: 0 = Off, 1 = On
+        """
+        self._sendOrder(b"RANGE %d,%d" % (self._output_channel, hrange))
+
+    def GetHeaterRange(self):
+        """
+        Query heater range enum int from the device
+        reutrns: (int): Depending on setup, for Outputs 1 and 2 in Current mode: 0 = Off, 1 = Low, 
+            2 = Medium, 3 = High
+            For Output 2 in Voltage mode: 0 = Off, 1 = On
+        """
+        return int(self._sendQuery(b'RANGE? %d' % (self._output_channel,)))
+
+    # Internal API functions
+    def _set_targetTemperature(self, value):
+        """
+        Setter for the targetTemperature VA
+        VA is in Celsius, but controller uses Kelvin
+        """
+        value = float(value)
+        self.SetSetpoint(value)
+        # Read back the new value from the device
+        svalue = self.GetSetpoint()
+        self.checkError()
+        if abs(value - svalue) >= 0.01:
+            logging.warning("Did not set new target temperature to %f", value)
+        return svalue
+
+    def _set_heating(self, value):
+        """
+        Setter for the heater range VA
+        """
+        value = int(value)
+        self.SetHeaterRange(value)
+        # Read back the new value from the device
+        svalue = self.GetHeaterRange()
+        self.checkError()
+        if svalue != value:
+            logging.warning("Did not set new heating range to %d", value)
+        return svalue
+
+    def _poll(self):
+        '''
+        This method runs in a separate thread and polls the device for the temperature
+        '''
+        try:
+            temp = self.GetSensorTemperature()
+            self.temperature._set_value(temp, force_write=True)
+            logging.debug(u"Lakeshore temperature: %f °C", self.temperature.value)
+            self.checkError()
+        except Exception as e:
+            # another exception. End running.
+            logging.exception(e)
```

Comment:
The comment says it ends running, but it actually catches the exception and keeps running => update the comment.

Also, `logging.exception()` already logs the exception string, so no need to pass it. Instead, say something like "Failed to read sensor temperature".